File: Modules/Recommendations/Models/Recommendation.php
<?php

namespace Modules\Recommendations\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Modules\Recommendations\Enums\RecommendationType;
use Modules\User\Models\SubstanceCategory;

class Recommendation extends Model
{
    protected $fillable = [
        'substance_category_id',
        'type',
        'name_ar',
        'name_en',
        'contact_info',
        'latitude',
        'longitude',
        'is_active',
    ];

    protected $casts = [
        'type' => RecommendationType::class,
        'is_active' => 'boolean',
    ];

    public function category(): BelongsTo
    {
        return $this->belongsTo(SubstanceCategory::class, 'substance_category_id');
    }
}

File: Solid.Api/Database/Entities/Recommendation.cs
namespace Solid.Api.Database.Entities;

public sealed class Recommendation
{
    public long Id { get; set; }

    public long? CreatedByUserId { get; set; }

    public long? SubstanceCategoryId { get; set; }

    public string Type { get; set; } = string.Empty;

    public string NameAr { get; set; } = string.Empty;

    public string NameEn { get; set; } = string.Empty;

    public string? ContactInfo { get; set; }

    public decimal? Latitude { get; set; }

    public decimal? Longitude { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public SubstanceCategory? SubstanceCategory { get; set; }

    public User? CreatedByUser { get; set; }
}
